### PR TITLE
Users/kmkumaran/multipbatchsupport

### PR DESCRIPTION
**Pull request by kmkumaran**
Added support for handling multiple batches for given SQL scripts.

### CODE REVIEW

File: Tasks/SqlDacpacDeploymentOnMachineGroup/GenerateSqlBatchFiles.ps1 [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    }
+       
+    #Send the line as is       
+    return $line, $null
+}
+
+function Test-IsSplitRequired
+{
+    param
+    (
```
Comment (Ajay-MS) [inline Tasks/SqlDacpacDeploymentOnMachineGroup/GenerateSqlBatchFiles.ps1]: Get-LinestoAppend and Test-IsSplitRequired have duplication of logic. Can Test-IsSplitRequired be dependent on Get-LinestoAppend

**Comment by asranja** [comment]
Please add few verbose statements.

File: Tasks/SqlDacpacDeploymentOnMachineGroup/GenerateSqlBatchFiles.ps1 [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        [string]$destPath,
+        [string]$batch
+    )
+    
+    $batchIndex = 1
+    $sqlFileName = Split-Path $sqlFilePath -Leaf
+    $reader = new-object System.IO.StreamReader($sqlFilePath)  
+    $ext = "sql"
+    $batchFileName = Get-NewBatchFileName -destPath $destPath -batch $batch -batchIndex $batchIndex -sqlFileName $sqlFileName -ext $ext
+
```
Comment (asranja) [inline Tasks/SqlDacpacDeploymentOnMachineGroup/GenerateSqlBatchFiles.ps1]: The batch file name is generated as 1.1.filename.sql.sql. Should it just be 1.1.filename.sql ?

File: Tasks/SqlDacpacDeploymentOnMachineGroup/Main.ps1 [hunk 3]
```diff
@@ -42,6 +42,25 @@ function Get-SingleFile
...
+{
+    param (
+        [string]$directoryName
+    )
+
+    if ((Test-Path -Path $directoryName) -eq $true) 
+    {
+        
+        Get-ChildItem -Path $directoryName -Force -Recurse |
+            Sort-Object -Property FullName -Descending |
```
[the remark this answers is not in the document]
Comment (kmkumaran) [reply Tasks/SqlDacpacDeploymentOnMachineGroup/Main.ps1]: Directory can be nested. This is to ensure we go in order.

File: Tasks/SqlDacpacDeploymentOnMachineGroup/GenerateSqlBatchFiles.ps1 (hunk 2 above)
Comment (kmkumaran) [reply Tasks/SqlDacpacDeploymentOnMachineGroup/GenerateSqlBatchFiles.ps1]: This still works. These are temporary batches. Hence, I can take this if it is really required.

**Pull request merged**